lib/src/nodes/config_node_simple_value.cc

```diff
@@ -0,0 +1,44 @@
+#include <internal/nodes/config_node_simple_value.hpp>
+#include <internal/config_exception.hpp>
+#include <internal/tokens.hpp>
+#include <internal/values/config_string.hpp>
+
+using namespace std;
+
+namespace hocon {
+
+    config_node_simple_value::config_node_simple_value(shared_token value) : _value(move(value)) { }
+
+    shared_token config_node_simple_value::get_token() const {
+        return _value;
+    }
+
+    token_list config_node_simple_value::get_tokens() const {
+        return token_list { _value };
+    }
+
+    unique_ptr<abstract_config_value> const& config_node_simple_value::get_value() const {
+        shared_ptr<value> value_token = dynamic_pointer_cast<value>(_value);
```

Comment:
I think I actually found it thanks to CLion suggesting methods. I typed "dynamic" and saw it, then googled it because I was curious. Seemed to be exactly what I wanted.